__tests__/retrieve_date__test.js

```diff
@@ -3,231 +3,305 @@ const sinon = require('sinon');
+        const period = datetimeUtils.retrievePeriod('last_3_days_including_current');
+        expect(period.start).to.equal('2017-03-17');
+        expect(period.end).to.equal('2017-03-19');
       });
-    });
 
-    it('should retrieve last_1_year_including_current', () => {
-      const period = datetimeUtils.retrievePeriod('last_1_year_including_current');
-      expect(period).to.deep.equal({
-        start: '2017-01-01',
-        end: '2017-12-31',
+      it('should last_2_weeks_including_current period be [06/03/2017 - 19/03/2017]', () => {
+        const period = datetimeUtils.retrievePeriod('last_2_weeks_including_current');
+        expect(period.start).to.equal('2017-03-06');
+        expect(period.end).to.equal('2017-03-19');
       });
-    });
 
-    afterAll(() => {
-      clock.restore();
-    });
-  });
+      it('should last_1_quarter_including_current period be [01/01/2017 - 19/03/2017]', () => {
+        const period = datetimeUtils.retrievePeriod('last_1_quarter_including_current');
+        expect(period.start).to.equal('2017-01-01');
+        expect(period.end).to.equal('2017-03-19');
+      });
 
-  describe('with utc baseDate to 1st of january 2015, 12:00am', () => {
-    const january2015 = 1420070400000;
-    it('should retrieve last_month', () => {
-      const period = datetimeUtils.retrievePeriod('last_month', january2015);
-      expect(period.start).to.equal('2014-12-01');
-      expect(period.end).to.equal('2014-12-31');
-    });
+      it('should last_1_year_including_current period be [01/01/2017 - 19/03/2017]', () => {
+        const period = datetimeUtils.retrievePeriod('last_1_year_including_current');
+        expect(period).to.deep.equal({
+          start: '2017-01-01',
+          end: '2017-03-19',
+        });
+      });
 
-    it('should retrieve this_month', () => {
-      const period = datetimeUtils.retrievePeriod('this_month', january2015);
-      expect(period.start).to.equal('2015-01-01');
-      expect(period.end).to.equal('2015-01-01');
-    });
+      it('should 2014-01-01_to_yesterday period be [01/01/2014 - 19/03/2017]', () => {
+        const period = datetimeUtils.retrievePeriod('2014-01-01_to_yesterday');
+        expect(period).to.eql({
+          start: '2014-01-01',
+          end: '2017-03-19',
+        });
+      });
 
-    it('should retrieve this_month', () => {
-      const period = datetimeUtils.retrievePeriod('this_year', january2015);
-      expect(period.start).to.equal('2015-01-01');
-      expect(period.end).to.equal('2015-12-31');
+      afterAll(() => {
+        clock.restore();
+      });
     });
 
-    it('should retrieve yesterday', () => {
-      const period = datetimeUtils.retrievePeriod('yesterday', january2015);
-      expect(period.start).to.equal('2014-12-31');
-      expect(period.end).to.equal('2014-12-31');
-    });
+    describe('with utc baseDate to 1st of january 2015, 12:00am', () => {
```

Comment:
That looks like a job for `context`

Reply:
There is no `context` function in jest

Reply:
Oh right I forgot we're using Jest there